decidim-core/app/middleware/decidim/current_organization.rb

```diff
@@ -17,6 +17,7 @@ def initialize(app)
     def call(env)
       organization = detect_current_organization(env)
       if organization
+        RequestStore.store[:decidim_current_organization] = organization
```

Comment:
This looks a lot like upcoming Rails 5.2's `Current` class, which I personally dislike because it keeps a global state. What about making the `ContentParsers` receive a new argument on initialization called `context`? This could be a hash, so we could send the organization there and any other information we might find useful in the future (for example, the resource we're parsing, or the space this resource lives in, etc).

What do you think? 😄 

I'd rather have this `context` than adding a new dependency to hold the global state, sincerely. Also, my method would avoid having to skip the rubocop checks in the tests, as you can set the context on test setup.

Reply:
I'd also prefer injecting the current organization instead of having this global state.